overlord/servicestate/service_control_test.go

```diff
@@ -407,12 +421,32 @@ func (s *serviceControlSuite) TestControlStopDisableMultipleInstruction(c *C) {
 
 	inst := &servicestate.Instruction{
 		Action:      "stop",
-		Names:       []string{"foo", "bar"},
+		Names:       []string{},
 		StopOptions: client.StopOptions{Disable: true},
 	}
 
-	chg := makeControlChange(c, st, inst, inf)
-	verifyControlTasks(c, chg.Tasks(), "stop", "disable", "snap.test-snap.foo.service", "snap.test-snap.bar.service")
+	chg := makeControlChange(c, st, inst, inf, []string{"foo", "bar"})
+	verifyControlTasks(c, chg.Tasks(), "stop", "disable",
+		[]string{"snap.test-snap.foo.service", "snap.test-snap.bar.service"},
+		[]string{})
+}
+
+func (s *serviceControlSuite) TestWithExplicitServices(c *C) {
+	st := s.state
+	st.Lock()
+	defer st.Unlock()
+
+	inf := s.mockTestSnap(c)
+
+	inst := &servicestate.Instruction{
+		Action: "start",
+		Names:  []string{"test-snap.foo", "test-snap.abc"},
+	}
```

Comment:
We should also have a test case for mixing explicit services and snap names (for same snap), e.g. `Names: []string{"test-snap.foo", "test-snap.abc", "test-snap"`}.

Reply:
Added one more test